front/src/applications/rollingStockEditor/consts.ts

```diff
@@ -1,47 +1,62 @@
 import { ConditionalEffortCurve, EffortCurve } from 'common/api/osrdEditoastApi';
 
-export type RollingStockParametersValues = {
-  [key: number | string]: {
-    name: string;
-    railjson_version: string;
-    version: number;
-    detail: string;
-    family: string;
-    grouping: string;
-    number: string;
-    reference: string;
-    series: string;
-    subseries: string;
-    type: string;
-    unit: string;
-    length: number;
-    mass: number;
-    max_speed: number;
-    startup_time: number;
-    startup_acceleration: number;
-    comfort_acceleration: number;
-    gamma_value: number;
-    inertia_coefficient: number;
-    loading_gauge: string;
-    rolling_resistance_A: number;
-    rolling_resistance_B: number;
-    rolling_resistance_C: number;
-    electrical_power_startup_time: number;
-    raise_pantograph_time: number;
-    default_mode: string;
-    modes: {
-      [key: string]: {
-        curves: ConditionalEffortCurve[];
-        default_curve: EffortCurve;
-        is_electric: boolean;
-      };
+type EffortCurves = {
+  modes: {
+    [key: string]: {
+      curves: ConditionalEffortCurve[];
+      defaultCurve: EffortCurve;
+      isElectric: boolean;
     };
   };
 };
 
+export type RollingStockParametersValues = {
+  [key: string]: string | number | null | EffortCurves;
+  railjsonVersion: string;
+  name: string;
+  detail: string;
+  family: string;
+  grouping: string;
+  number: string;
+  reference: string;
+  series: string;
+  subseries: string;
+  type: string;
+  unit: string;
+  length: number;
+  mass: number;
+  maxSpeed: number;
+  startupTime: number;
+  startupAcceleration: number;
+  comfortAcceleration: number;
+  gammaValue: number;
+  inertiaCoefficient: number;
+  loadingGauge: 'G1' | 'G2' | 'GA' | 'GB' | 'GB1' | 'GC' | 'FR3.3' | 'FR3.3/GB/G2' | 'GLOTT';
+  rollingResistanceA: number;
+  rollingResistanceB: number;
+  rollingResistanceC: number;
+  electricalPowerStartupTime: number | null;
+  raisePantographTime: number | null;
+  defaultMode: string;
+  effortCurves: EffortCurves;
+};
+
+export type SchemaProperty = {
+  title: string;
+  type: string;
+  required: boolean;
+  group: string;
+  side: string;
+  format?: string;
+  enum?: string[];
+  min?: number;
+  max?: number;
+  unit?: string;
+  units?: string[];
+};
+
 export enum Metadata {
```

Comment:
As Metadata is exported, we can have access to this enum everywhere in the application, and the name is not very precise here. You should also transform this into an array I think

```suggestion
export enum RollingStockEditorMetadata {
```

Reply:
done